Unix/base/interaction.h

```diff
@@ -44,6 +44,7 @@ typedef struct _InteractionOpenParams
     Interaction*    interaction;
     void*           callbackData;   // Optional callbackData (usually the object being called back)
     Message*        msg;            // Optional initial message, if not NULL receiver SHOULD Ack when is ready
+    MI_Char*        errmsg;            // Optional error message, used by Strand_FailOpenWithResult
```

Comment:
This structure is used by a lot of components, and yet it is only used in one very specific scenario. By changing the call stack of just a couple of functions you can eliminate this such that:
ProvMgr_OpenCallback calls into ProvMgr_NewRequest with a new out param for the error, 
ProvMgr_NewRequest then calls into the different _Handle*Req functions with new out param.
Then, back in ProvMgr_OpenCallback I would call into Strand_FailOpenWithResult with a new _In_opt_ parameter that is the errorMsg following the current result parameter. That can then pass the optional errorMsg into the (*callback) function call to add the string.
The result is we can revert this specific change.